Template/Tag/MatomoTag.php

```diff
@@ -205,7 +205,46 @@ public function getParameters()
                     }
                     return $value;
                 };
-            })
+            }),
+            $this->makeSetting('customDimensions', [], FieldConfig::TYPE_ARRAY, function (FieldConfig $field) {
+                $field->title = Piwik::translate('TagManager_MatomoConfigurationMatomoCustomDimensionsTitle');
+                $field->description = Piwik::translate('TagManager_MatomoTagCustomDimensionsDescription');
+                $field->validate = function ($value) {
+                    if (empty($value)) {
+                        return;
+                    }
+                    if (!is_array($value)) {
+                        throw new \Exception(Piwik::translate('TagManager_MatomoConfigurationMatomoCustomDimensionsException'));
+                    }
+                };
+
+                $field->transform = function ($value) {
+                    if (empty($value) || !is_array($value)) {
+                        return [];
+                    }
+                    $withValues = [];
+                    foreach ($value as $dim) {
+                        if (!empty($dim['index']) && !empty($dim['value'])) {
+                            $withValues[] = $dim;
+                        }
+                    }
+
+                    return $withValues;
+                };
+
+                $field->uiControl = FieldConfig::UI_CONTROL_MULTI_TUPLE;
+                $field1 = new FieldConfig\MultiPair('Index', 'index', FieldConfig::UI_CONTROL_TEXT);
+                $field1->customFieldComponent = self::FIELD_VARIABLE_COMPONENT;
+                $field2 = new FieldConfig\MultiPair('Value', 'value', FieldConfig::UI_CONTROL_TEXT);
+                $field2->customFieldComponent = self::FIELD_VARIABLE_COMPONENT;
+                $field->uiControlAttributes['field1'] = $field1->toArray();
+                $field->uiControlAttributes['field2'] = $field2->toArray();
+            }),
+            $this->makeSetting('areCustomDimensionsSticky', false, FieldConfig::TYPE_BOOL, function (FieldConfig $field) {
```

Comment:
@snake14 Without setting this config, the dimensions are still persisting

![Screenshot from 2024-09-27 11-32-00](https://github.com/user-attachments/assets/35042191-51e3-4584-8156-73a35ec91027)



![Screenshot from 2024-09-27 11-35-03](https://github.com/user-attachments/assets/843810e3-19f2-45e3-bbf6-cee81564d80a)

Reply:
@AltamashShaikh I'm unable to reproduce. When I have the checkbox disabled, the custom dimension is only used when the specific tag is triggered. For example, I have the custom dimension configured for a tag using an image click trigger. When I click other elements, the dimension isn't present in the request, clicking the image includes the dimension, and the dimension is gone again when I click a different element.

Reply:
@AltamashShaikh I just recorded my demo video (see Slack channel) and it worked as expected.

Reply:
@snake14 If it works we can merge it